javascript/react/src/hooks/manageWorkflow.ts

```diff
@@ -126,11 +127,17 @@ export function useWorkflowOutputListener(
 	useEffect(
 		() =>
 			setOutput &&
-			workflow.onChange(
-				() =>
-					setOutput(workflow.toArray().filter(t => !!t) as TableContainer[]),
-				true,
-			),
+			workflow.onChange(() => {
+				const outputs = (
+					workflow.toArray().filter(t => !!t) as TableContainer[]
+				).map(table => {
+					if (!table.metadata) {
+						table.metadata = introspect(table.table!, true)
```

Comment:
Is this the right place to wire in introspection? This kind of implies that this hook _must_ be used to have metadata. I wonder if we should wire metadata into tables within the Worfklow class.

Reply:
It is not, but once we have our table-schema-generating flow we can opt in and inject it. In a case like this we may need some option to generate schemas automatically on derived tables, since it won't quite be in the user flow (like "Create codebook") is.